### PR TITLE
Fix bug in PixelThreshold for single input

### PR DESCRIPTION
**Pull request by aufieroma12**
Calling `.generate()` on a batch of length one, a `TypeError` is raised on line 193. This is because `np.squeeze(y)` returns a 0-dimensional array when `y.shape == (1, 1)`. Changing it to `y.flatten()` solves this issue by returning a 1-dimensional array in this case, while not changing the behavior in the case when the batch has more than one input.

Signed-off-by: Massimo Aufiero <aufieroma12@gmail.com>

Fixes # (issue)

## Type of change

Please check all relevant options.

- [ ] Improvement (non-breaking)
- [x] Bug fix (non-breaking)
- [ ] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Testing

Please describe the tests that you ran to verify your changes. Consider listing any relevant details of your test configuration.

- [x] Ran Pixel Attack on input array of shape `(1, 3, 224, 224)` (this did not work before, and it does not)
- [x] Ran Pixel Attack on input array of shape `(3, 3, 224, 224)` (this still works as before)

**Test Configuration**:
- OS
- Python version
- ART version or commit number
- TensorFlow / Keras / PyTorch / MXNet version

# Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by beat-buesser — APPROVED** [review]
Hi @aufieroma12 Thank you very much for your pull request and fixing support for single samples in PixelThreshold! The change looks good to me. If you have time, you could add a new unit test for the single sample case?

**Comment by aufieroma12** [comment]
> Hi @aufieroma12 Thank you very much for your pull request and fixing support for single samples in PixelThreshold! The change looks good to me. If you have time, you could add a new unit test for the single sample case?

Sure thing, just added one!

**Comment by beat-buesser** [comment]
That's great! Thank you very much.

File: tests/attacks/test_pixel_attack.py [hunk 1]
```diff
@@ -91,6 +91,15 @@ def test_4_pytorch_mnist(self):
...
         self._test_attack(classifier, x_test, self.y_test_mnist, False)
 
+    def test_8_pytorch_mnist_single_sample(self):
+        """
+        Test with the PyTorchClassifier on a single sample. (Untargeted Attack)
+        :return:
+        """
+        x_test = np.reshape(self.x_test_mnist[0], (1, 1, 28, 28)).astype(np.float32)
+        classifier = get_image_classifier_pt()
+        self._test_attack(classifier, x_test, self.y_test_mnist, False)
```
Comment (beat-buesser) [inline tests/attacks/test_pixel_attack.py:101]:
```suggestion
        x_test = np.reshape(self.x_test_mnist[1], (1, 1, 28, 28)).astype(np.float32)
        classifier = get_image_classifier_pt()
        self._test_attack(classifier, x_test, self.y_test_mnist[[1]], False)
```

Comment (beat-buesser) [reply tests/attacks/test_pixel_attack.py:101]: Make test more robust.

**Pull request merged**